app/server/middleware/search.js

```diff
@@ -0,0 +1,32 @@
+import axios from 'axios';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const searchArticles = (request, response, next) => {
+  // only searches a single source, rather than an
+  // array of sources
+  // also doesn't allow sortBy filtering yet, as some
+  // sources don't support all types of filtering
+  const { source } = request.query;
+
+  const search = `https://newsapi.org/v1/articles?apiKey=${process.env.NEWS_KEY}&source=${source}`;
+
+  axios.get(search)
+    .then((data) => {
+      const articles = data.data.articles.map((article) => {
+        const newArticle = article;
+        newArticle.source = source;
+        return article;
```

Comment:
maybe I'm reading this wrong, but do we not want to return newArticle?

Reply:
You are 100% right. I'd made the change from article to newArticle to keep the AirBnB linter happy and forgot to return the newArticle. I've fixed that.